src/components/CustomTable/CustomTable.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import { Box, Table, TableBody, TableCell, TableContainer, TablePagination, TableRow } from '@mui/material';
+import { ChangeEvent, MouseEvent, useEffect } from 'react';
+import Checkbox from '@mui/material/Checkbox';
+import { CustomTableProps } from '../../common/types/TableTypes.types';
+import { FetchError } from '../../errors/FetchError';
+
+export function CustomTable<Type extends { id: number }>(props: CustomTableProps<Type>) {
+  const { head, toolbar, fetchUrl, customTableStatefulVariables } = props;
```

Comment:
this constraint is too strict. We should use `Record<PropertyKey, unknown>`

Reply:
Replaced the constraint in [92390f3](https://github.com/dorm-social-room-manager/dsrm-frontend/pull/38/commits/92390f35bc955ce6069d373c264463c3e29f685a)